src/base/static/components/organisms/main-map.tsx

```diff
@@ -51,9 +50,60 @@ import MapMeasurementOverlay from "../organisms/map-measurement-overlay";
 
 import { Mixpanel } from "../../utils/mixpanel";
 import { FlyToInterpolator } from "react-map-gl";
+import produce from "immer";
 
 const transitionInterpolator = new FlyToInterpolator();
 
+const UPDATE_VIEWPORT = "update-viewport";
+const mapViewportReducer = (
+  state: MapViewport,
+  action: {
+    type: string;
+    payload: { viewport: MapViewportDiff; scrollZoomAroundCenter: boolean };
+  },
+): MapViewport =>
+  produce(state, draft => {
+    switch (action.type) {
+      case UPDATE_VIEWPORT:
+        if (!isNaN(Number(action.payload.viewport.zoom))) {
```

Comment:
Is it necessary to run these `isNaN` checks every time? I guess I'm wondering under what circumstances the viewport state would be non-numeric.

Reply:
Good point - I can replace these checks with something like `action.payload.viewport.zoom !== undefined`. Note that checks like `if (!action.payload.viewport.zoom)` won't work, because they fail when the value is 0.